src/modules/mapper/generic.mapper.ts

```diff
@@ -24,5 +21,10 @@ export class GenericMapper {
                 .getOwnPropertyNames(domainModel)
                 .map(property => (property.replace('_', '')) );
     }
+
+    private static ensureDomainHasGetterForProperty(domainModel: Domain, property: string): void {
```

Comment:
Maybe, not now because we don't have any use case yet, but in the future, every domain class must return the custom list of properties that want to be accesible. As getOwnPropertyNames returns everything, maybe we could have a custom implementation not using Object.getOwnPropertyNames and having domain.getOwnPropertyNames and returning a list. Something like return ["publicProperty1", "publicProperty2"]...
As we still don't have the use case, I think it's perfectly valid

Reply:
Ok, it sounds reasonable.
May be a custom method returning only public properties will be better.